src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -34,6 +34,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
   val superclass = builder.superclass
   val superclassConstructorParameters = builder.superclassConstructorParameters.toImmutableList()
   val superinterfaces = builder.superinterfaces.toImmutableList()
+  val delegateSuperInterfaces = builder.delegateSuperInterfaces.toImmutableMap()
```

Comment:
Not sure I like the name. Since that map maps superinterfaces to their delegates, I think a better name would be `superinterfaceDelegates`, or `delegatedSuperinterfaces`.